### PR TITLE
Update workflow and docs for releasing Spark operator

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

Update release workflow:

- The integration test workflow is triggered by pull requests and pushes to `main` branch and `release-*` branch.
- After a `v*.*.*` tag is pushed, a workflow will be triggered to build and push docker images to Docker Hub.
- After a `v*.*.*` tag is pushed, a workflow will be triggered to create a draft release.
- Afeter a release is published, a workflow will be triggered to update Helm repo index to publish charts.
- Add docs for releasing spark operator.

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [x] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [x] I have updated documentation accordingly.
- [ ] I have added tests that prove my changes are effective or that my feature works.
- [ ] Existing unit tests pass locally with my changes.

### CODE REVIEW

File: docs/release.md
Comment (andreyvelich) [inline docs/release.md:66]: I am wondering if we can automate the tag pushing somehow ?
For example, the GitHub workflow will push tag to the upstream when controller image build was complete.
 This workflow can be triggered if VERSION file has been updated

**Review by andreyvelich — COMMENTED** [review]
Thank you for this @ChenYi015!

File: docs/release.md
Comment (ChenYi015) [reply docs/release.md]: @andreyvelich Sounds good! I had added a workflow for pushing tags when VERSION file changes.

**Comment by ChenYi015** [comment]
@yuchaoran2011 PTAL.

File: VERSION
Comment (andreyvelich) [inline VERSION:1]: @ChenYi015 Improving the release process is out of scope of this PR: https://github.com/kubeflow/spark-operator/issues/1975 ?
E.g. that we should keep the same version for Spark Operator image and Helm charts release.

Comment (ChenYi015) [reply VERSION:1]: > @ChenYi015 Improving the release process is out of scope of this PR: #1975 ? E.g. that we should keep the same version for Spark Operator image and Helm charts release.

@andreyvelich I have updated the doc to keep the versions the same when creating a git tag.

```
VERSION=$(cat VERSION)
sed -i "s/^version.*/version: ${VERSION}/" charts/spark-operator-chart/Chart.yaml
sed -i "s/^appVersion.*/appVersion: ${VERSION}/" charts/spark-operator-chart/Chart.yaml
```

Comment (andreyvelich) [reply VERSION:1]: @ChenYi015 Since we keep Chart and Image version the same, why do we need `appVersion` parameter in Helm Charts: https://github.com/kubeflow/spark-operator/blob/a8ee09a3a6a4cf3140c0ea8b2c3eac7b057c1090/charts/spark-operator-chart/Chart.yaml#L5

For example, KServe has only `version` [in the Chart file ](https://github.com/kserve/kserve/blob/master/charts/kserve-resources/Chart.yaml#L3)

Comment (ChenYi015) [reply VERSION:1]: @andreyvelich `appVersion` is indeed an optional field, but it can give information about the operator version, when helm users search charts like this, the `APP VERSION` will be shown:

```bash
$ helm search repo spark-operator/spark-operator                                         
NAME                            CHART VERSION   APP VERSION             DESCRIPTION                                   
spark-operator/spark-operator   2.0.0           2.0.0                   A Helm chart for Spark on Kubernetes operator
``` 

And if we removed it, it will be like this and may cause some confusion:

```bash
$ helm search repo spark-operator/spark-operator                                         
NAME                            CHART VERSION   APP VERSION             DESCRIPTION                                   
spark-operator/spark-operator   2.0.0                                   A Helm chart for Spark on Kubernetes operator
```

Besides, we will not need to modify the chart templates if we choose to keep the `appVersion` field, as all resources created by the chart will have `appVersion` labels.

Comment (andreyvelich) [reply VERSION:1]: Do you think it is necessary for users since Chart version is always equal to the Image version ?

Comment (ChenYi015) [reply VERSION:1]: Though they are the same, I think it is better to keep the `appVersion` field as it provides extra metadata information.

**Comment by ChenYi015** [comment]
@andreyvelich Do you think it is ok? After this PR and #2072  are merged, we can make a `v2.0.0-rc.0` release.

**Comment by andreyvelich** [comment]
> @andreyvelich Do you think it is ok? After this PR and #2072 are merged, we can make a `v2.0.0-rc.0` release.

Yeah, but we need to cherry-pick all of the required PRs into `release-2.0` branch. cc @vara-bonthu @yuchaoran2011

**Review by vara-bonthu — APPROVED** [review]
@ChenYi015 
Thanks for the PR and it's going to streamline the release process. 
Once this PR is merged and all the approvers are satisfied, we can proceed to create the `v2.0.0-rc.0` release. We need to ensure we cherry pick changes merged to main until the new release branch is merged.

/approve

**Comment by ChenYi015** [comment]
@yuchaoran2011 Could you also review this PR? I have tested this workflow and it works fine.

**Review by yuchaoran2011 — APPROVED** [review]
/lgtm

**Pull request merged**
